### PR TITLE
Colorable Dresses

### PR DESCRIPTION
**Pull request by atteria**
Adds seven ~~re~~colorable dresses to the loadout.

I let these sprites sit for far too long. I hope people enjoy them.

image

### CODE REVIEW

**Comment by atteria** [comment]
!review

File: code/modules/client/preference_setup/loadout/items/uniform.dm [hunk 1]
```diff
@@ -434,6 +434,24 @@
...
+	dress_recolorable["strapless midi dress"] = /obj/item/clothing/under/dress/recolorable
+	dress_recolorable["sleeveless A-line dress"] = /obj/item/clothing/under/dress/recolorable/sleeveless
+	dress_recolorable["longsleeve A-line dress"] = /obj/item/clothing/under/dress/recolorable/longsleeve
+	dress_recolorable["evening gown"] = /obj/item/clothing/under/dress/recolorable/evening_gown
+	dress_recolorable["tea-length dress"] = /obj/item/clothing/under/dress/recolorable/tea_dress
+	dress_recolorable["open-shoulder dress"] = /obj/item/clothing/under/dress/recolorable/open_shoulder
+	dress_recolorable["asymmetric dress"] = /obj/item/clothing/under/dress/recolorable/asymmetric
+
+	gear_tweaks += new /datum/gear_tweak/path(dress_recolorable)
+
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/items/uniform.dm:454]:
```suggestion
/datum/gear/uniform/miscellaneous/dress_colorable
	display_name = "dress selection (colorable)"
	path = /obj/item/clothing/under/dress/colorable
	flags = GEAR_HAS_NAME_SELECTION | GEAR_HAS_DESC_SELECTION | GEAR_HAS_COLOR_SELECTION

/datum/gear/uniform/miscellaneous/dress_colorable/New()
	..()
	var/list/dress_colorable = list()
	dress_colorable["strapless midi dress"] = /obj/item/clothing/under/dress/colorable
	dress_colorable["sleeveless A-line dress"] = /obj/item/clothing/under/dress/colorable/sleeveless
	dress_colorable["longsleeve A-line dress"] = /obj/item/clothing/under/dress/colorable/longsleeve
	dress_colorable["evening gown"] = /obj/item/clothing/under/dress/colorable/evening_gown
	dress_colorable["tea-length dress"] = /obj/item/clothing/under/dress/colorable/tea_dress
	dress_colorable["open-shoulder dress"] = /obj/item/clothing/under/dress/colorable/open_shoulder
	dress_colorable["asymmetric dress"] = /obj/item/clothing/under/dress/colorable/asymmetric

	gear_tweaks += new /datum/gear_tweak/path(dress_colorable)

```

File: code/modules/clothing/under/dresses.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/obj/item/clothing/under/dress/recolorable/open_shoulder
+	name = "open-shoulder dress"
+	icon_state = "open_shoulder_dress"
+	item_state = "open_shoulder_dress"
+
+/obj/item/clothing/under/dress/recolorable/asymmetric
+	name = "asymmetric dress"
+	icon_state = "asymm_dress"
+	item_state = "asymm_dress"
```
Comment (SleepyGemmy) [inline code/modules/clothing/under/dresses.dm:42]:
```suggestion
/obj/item/clothing/under/dress/colorable
	name = "strapless midi dress"
	desc = "A one-piece fashionable garment."
	icon = 'icons/clothing/under/uniforms/colorable_dresses.dmi'
	icon_state = "dress"
	item_state = "dress"
	body_parts_covered = UPPER_TORSO | LOWER_TORSO
	contained_sprite = TRUE

/obj/item/clothing/under/dress/colorable/sleeveless
	name = "sleeveless A-line dress"
	icon_state = "shortsleeve_dress"
	item_state = "shortsleeve_dress"

/obj/item/clothing/under/dress/colorable/longsleeve
	name = "longsleeve A-line dress"
	icon_state = "longsleeve_dress"
	item_state = "longsleeve_dress"
	body_parts_covered = UPPER_TORSO | LOWER_TORSO | ARMS

/obj/item/clothing/under/dress/colorable/evening_gown
	name = "evening gown"
	icon_state = "evening_gown"
	item_state = "evening_gown"
	body_parts_covered = UPPER_TORSO | LOWER_TORSO | LEGS

/obj/item/clothing/under/dress/colorable/tea_dress
	name = "tea-length dress"
	icon_state = "tea_dress"
	item_state = "tea_dress"
	body_parts_covered = UPPER_TORSO | LOWER_TORSO | LEGS

/obj/item/clothing/under/dress/colorable/open_shoulder
	name = "open-shoulder dress"
	icon_state = "open_shoulder_dress"
	item_state = "open_shoulder_dress"

/obj/item/clothing/under/dress/colorable/asymmetric
	name = "asymmetric dress"
	icon_state = "asymm_dress"
	item_state = "asymm_dress"
```

File: icons/clothing/under/uniforms/recolorable_dresses.dmi
Comment (SleepyGemmy) [inline icons/clothing/under/uniforms/recolorable_dresses.dmi:1]: rename from `recolorable_dresses` to `colorable_dresses`.

File: html/changelogs/atteria-dresses.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+author: atteria
+
+delete-after: True
+
+changes:
+  - imageadd: "Added seven recolorable dresses to the loadout."
```
Comment (SleepyGemmy) [inline html/changelogs/atteria-dresses.yml:6]:
```suggestion
  - imageadd: "Added seven colorable dresses to the loadout."
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]

**Pull request merged**